src/base/bittorrent/session.cpp

```diff
@@ -3816,22 +3823,18 @@ bool Session::addMoveTorrentStorageJob(TorrentHandleImpl *torrent, const QString
         if (iter != m_moveStorageQueue.end()) {
             // remove existing inactive job
             m_moveStorageQueue.erase(iter);
+            LogMsg(tr("\"%1\" has cancelled to move storage from \"%2\" to \"%3\".").arg(torrent->name(), currentLocation, iter->path));
         }
     }
 
-    QString currentLocation = torrent->actualStorageLocation();
-    if (!m_moveStorageQueue.isEmpty() && (m_moveStorageQueue.first().torrentHandle == torrentHandle)) {
-        // if there is active job for this torrent consider its target path as current location
-        // of this torrent to prevent creating meaningless job that will do nothing
-        currentLocation = m_moveStorageQueue.first().path;
-    }
-
-    if (QDir {currentLocation} == QDir {newPath})
+    if (QDir {currentLocation} == QDir {newPath}) {
```

Comment:
Can't we just compare them in QString type?

Reply:
QDir performs some additional checks. Don't forget that the same location can be pointed by different path strings.

Reply:
>QDir performs some additional checks. 

Personally I doubt what kind of useful checks it would perform... I suspect we really need to normalize it `QDir::absolutePath()` manually for it to be useful.
And I noticed this is just borrowed from old code, so never mind.